src/object.rs

```diff
@@ -414,4 +419,60 @@ impl<T: IsA<Object>> ObjectExt for T {
             (Some(ptype), Some(type_)) => ptype == type_,
         }
     }
+
+    fn connect<'a, N, F>(&self, signal_name: N, after: bool, callback: F) -> Result<u64, BoolError>
+        where N: Into<&'a str>, F: Fn(&[Value]) -> Option<Value> + Send + Sync + 'static {
+        let signal_name: &str = signal_name.into();
+
+        unsafe {
+            let obj = self.to_glib_none().0;
+            let klass = (*obj).g_type_instance.g_class as *mut gobject_ffi::GTypeClass;
+            let type_ = (*klass).g_type;
```

Comment:
That said, it might make sense to extract the above code into a new ObjectExt::get_type() function. Or what do you think?